RecoEgamma/EgammaHLTProducers/interface/EgammaHLTFilteredObjProducer.h

```diff
@@ -0,0 +1,174 @@
+#ifndef RECOEGAMMA_EGAMMAHLTPRODUCERS_EGAMMAHLTFILTEREDOBJPRODUCER_H
+#define RECOEGAMMA_EGAMMAHLTPRODUCERS_EGAMMAHLTFILTEREDOBJPRODUCER_H
+
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h" 
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateFwd.h"
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateIsolation.h"
+
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+namespace {
+  template<typename T> edm::Handle<T> getHandle(const edm::Event& event,const edm::EDGetTokenT<T>& token)
+  {
+    edm::Handle<T> handle;
+    event.getByToken(token,handle);
+    return handle;
+  }
+
+}
+
+template <typename OutCollType>
+class EgammaHLTFilteredObjProducer : public edm::stream::EDProducer<>{
+public:  
+  class SelectionCut {
+  private:
+    struct CutValues {
+      float cut;
+      float cutOverE;
+      float cutOverE2;
+      bool useEt;
+      std::function<bool(float,float)> compFunc;
+
+      CutValues(const edm::ParameterSet& pset):
+	cut(pset.getParameter<double>("cut")),
+	cutOverE(pset.getParameter<double>("cutOverE")),
+	cutOverE2(pset.getParameter<double>("cutOverE2")),
+	useEt(pset.getParameter<bool>("useEt")),
+	compFunc(std::less<float>()) {}
+    
+      bool operator()(const reco::RecoEcalCandidate& cand,float value)const{
+	float energy = useEt ? cand.et() : cand.energy();
+	return compFunc(value,cut) || compFunc(value/energy,cutOverE) || 
+	  compFunc(value/energy/energy,cutOverE2);
```

Comment:
it would be more effective to test ```compFunc(value,cut)  ``` first and only after that get the energy. Also, computing ```energyInv = useEt ? 1.f/cand.et() : 1.f/cand.energy();```